torchaudio/functional/filtering.py

```diff
@@ -642,6 +642,34 @@ def equalizer_biquad(
     return biquad(waveform, b0, b1, b2, a0, a1, a2)
 
 
+def filtfilt(
+    waveform: Tensor,
+    a_coeffs: Tensor,
+    b_coeffs: Tensor,
+    clamp: bool = True,
+) -> Tensor:
+    r"""Perform an IIR filter forward and backward to a waveform.
+
+    Inspired by https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.filtfilt.html
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of ``(..., time)``.  Must be normalized to -1 to 1.
+        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(n_order + 1)``.
+            Lower delays coefficients are first, e.g. ``[a0, a1, a2, ...]``.
+            Must be same size as b_coeffs (pad with 0's as necessary).
+        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(n_order + 1)``.
+            Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
+            Must be same size as a_coeffs (pad with 0's as necessary).
+        clamp (bool, optional): If ``True``, clamp the output signal to be in the range [-1, 1] (Default: ``True``)
+
+    Returns:
+        Tensor: Waveform with dimension of ``(..., time)``.
+    """
+    forward_filtered = lfilter(waveform, a_coeffs, b_coeffs, clamp)
+    backward_filtered = lfilter(forward_filtered.flip(-1), a_coeffs, b_coeffs, clamp).flip(-1)
```

Comment:
I moved the ``.flip(-1)`` to the end to make it more consistent 🙂
```suggestion
    forward_filtered = lfilter(waveform, a_coeffs, b_coeffs, clamp).flip(-1)
    backward_filtered = lfilter(forward_filtered, a_coeffs, b_coeffs, clamp).flip(-1)
```

Reply:
i feel like the semantics of the original make more sense — `forward_filtered = lfilter(waveform, a_coeffs, b_coeffs, clamp)` is indeed the forward-filtered waveform

Reply:
Yea I'm convinced 🙂. Yours is better than scipy impl